### PR TITLE
Codegen update 9

### PR DESCRIPTION
**Pull request by pawelprazak**
[Fix codegen for 'secrets' test schema](https://github.com/VirtusLab/besom/commit/1398b164c23ee4d194ecb27ec486181008dce60a)

### CODE REVIEW

File: codegen/src/CodeGen.scala [hunk 1]
```diff
@@ -466,7 +466,7 @@ class CodeGen(implicit providerConfig: Config.ProviderConfig, typeMapper: TypeMa
 
     // the type has to match Pulumi's resource type schema, e.g. kubernetes:core/v1:Pod
     // please make sure, it contains 'index' instead of empty module part if needed
-    val typ = Lit.String(typeToken)
+    val typ = Lit.String(typeToken.sanitized.asString)
```
Comment (prolativ) [inline codegen/src/CodeGen.scala:469]: Are we sure it's OK to modify the original unparsed token in any way?

File: codegen/src/CodeGen.scala [hunk 2]
```diff
@@ -459,7 +466,7 @@ class CodeGen(implicit providerConfig: Config.ProviderConfig, typeMapper: TypeMa
...
```
Comment (prolativ) [inline codegen/src/CodeGen.scala:469]: Are we sure it's OK to modify the original unparsed token in any way?

Comment (pawelprazak) [reply codegen/src/CodeGen.scala:469]: It is necessary, according to our own SDK validation, I'm not 100% sure the engine requires it. CC @lbialy

**Comment by pawelprazak** [comment]
Rebased on develop

File: codegen/src/CodeGen.scala (hunk 2 above)
Comment (pawelprazak) [reply codegen/src/CodeGen.scala:469]: I have an idea how to correct it. Coming soon.

**Comment by pawelprazak** [comment]
closed in favor of https://github.com/VirtusLab/besom/pull/270

**Pull request closed without merging**